server.go

```diff
@@ -1853,78 +1702,9 @@ out:
 		if cfg.SimNet {
```

Comment:
This section can be removed since it no longer applies here, however the behavior needs to be retained in connection manager and currently it has not.

In fact, you can see this issue when running with the `--simnet` flag:

```
$ btcd --simnet -ddebug

21:34:24 2016-10-20 [DBG] CMGR: Attempting connection:
21:34:24 2016-10-20 [DBG] CMGR: Connection failed : dial tcp: missing address
21:36:26 2016-10-20 [DBG] CMGR: Attempting connection:
21:36:26 2016-10-20 [DBG] CMGR: Attempting connection:
21:36:26 2016-10-20 [DBG] CMGR: Attempting connection:
21:36:26 2016-10-20 [DBG] CMGR: Attempting connection:
21:36:26 2016-10-20 [DBG] CMGR: Attempting connection:
21:36:26 2016-10-20 [DBG] CMGR: Attempting connection:
21:36:26 2016-10-20 [DBG] CMGR: Attempting connection:
21:36:26 2016-10-20 [DBG] CMGR: Connection failed : dial tcp: missing address
21:36:26 2016-10-20 [DBG] CMGR: Connection failed : dial tcp: missing address
```

The key point of the simuation test network is that it is only intended to connect to specified peers and actively avoids advertising and connecting to discovered peers to prevent it from becoming a public test network.  You can see this throughout the code where it avoids adding any address to the address manager, does not provide any peers in response to a `getaddr` message, and ignores any addresses received via an `addr` message.  That means the connection manager should only ever connect to peers specified via `--addpeer` or `--connect` when on simnet, and, due to that fact, `--addpeer` is effectively identical to `--connect` on simnet.

Luckily, the fix should be as simple as:

```diff
	// Only setup a function to return new addresses to connect to when
-	// not running in connect-only mode.
+	// not running in connect-only mode.  The simulation network is always
+	// in connect-only mode since it is only intended to connect to
+	// specified peers and actively avoid advertising and connecting to
+	// discovered peers in order to prevent it from becoming a public test
+	// network.
	var newAddressFunc connmgr.AddressFunc
-	if len(cfg.ConnectPeers) == 0 {
+	if !cfg.SimNet && len(cfg.ConnectPeers) == 0 {
```